### PR TITLE
Add array overloads to IInsertCommand and ISelectCommand

### LINKED ISSUE #85: Cannot pass an array of column names to Columns method
## Expected Behavior
It should be possible to pass an array of column names to `Columns` method while inserting a row.

## Actual Behavior
Passing an array is not possible:
```
Argument 1: cannot convert from 'string[]' to 'Flepper.QueryBuilder.SqlColumn'
```
and `SqlColumn` constructor is internal so it's not possible to create an array of that too.

## Steps to Reproduce the Problem

Create an array of column names and pass it to the `Columns` method:
```
var columnNames = new[] {"Column1", "Column2", "Column3"};

Connection.Insert().Into("Table1").Columns(columnNames).Values("Value1", "Value2", "Value3").Execute();
```

### PR DESCRIPTION
**Pull request by ctyar**
Fixes #85

### CODE REVIEW

File: Flepper.QueryBuilder.DapperExtensions/Extensions/DbConnectionExtensions.cs [hunk 1]
```diff
@@ -23,13 +23,22 @@ public static ISelectCommand Select(this IDbConnection dbConnection)
...
-        /// <typeparam name="T">Object</typeparam>
-        /// <returns></returns>
-        public static ISelectCommand Select<T>(this IDbConnection dbConnection) where T : class
+	    /// <summary>
+	    /// Create Select Command
+	    /// </summary>
+	    /// <param name="columns">Columns name</param>
+	    /// <param name="dbConnection">DbConnection Instance</param>
+	    /// <returns></returns>
+	    public static ISelectCommand Select(this IDbConnection dbConnection, string[] columns)
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder.DapperExtensions/Extensions/DbConnectionExtensions.cs]: Change `string[]`  for `params string[]`

File: Flepper.QueryBuilder/Commands/InsertCommand.cs [hunk 2]
```diff
@@ -17,7 +17,13 @@ public IInsertIntoCommand Columns(params SqlColumn[] columns)
             return this;
         }
 
-        public IValuesOperator WithScopeIdentity()
+	    public IInsertIntoCommand Columns(string[] columns)
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/InsertCommand.cs]: Change `string[]`  for `params string[]`

File: Flepper.QueryBuilder/Commands/Interfaces/IInsertIntoCommand.cs [hunk 3]
```diff
@@ -11,5 +11,12 @@ public interface IInsertIntoCommand : IQueryCommand
...
         /// <returns></returns>
         IInsertIntoCommand Columns(params SqlColumn[] columns);
-    }
+
+		/// <summary>
+		/// Columns contract
+		/// </summary>
+		/// <param name="columns"></param>
+		/// <returns></returns>
+		IInsertIntoCommand Columns(string[] columns);
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/Interfaces/IInsertIntoCommand.cs]: Change `string[]`  for `params string[]`

File: Flepper.QueryBuilder/Commands/Interfaces/ISelectCommand.cs [hunk 4]
```diff
@@ -21,11 +21,18 @@ public interface ISelectCommand : IQueryCommand
...
-        /// Select Command Contract
-        /// </summary>
-        /// <returns></returns>
-        ISelectCommand SelectCommand<T>() where T : class;
+	    /// <summary>
+	    /// Select Command Contract
+	    /// </summary>
+	    /// <param name="columns"></param>
+	    /// <returns></returns>
+	    ISelectCommand SelectCommand(string[] columns);
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/Interfaces/ISelectCommand.cs]: Change `string[]`  for `params string[]`

File: Flepper.QueryBuilder/Commands/SelectCommand.cs [hunk 5]
```diff
@@ -24,8 +24,13 @@ public ISelectCommand SelectCommand(params SqlColumn[] columns)
             return this;
         }
 
+	    public ISelectCommand SelectCommand(string[] columns)
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/SelectCommand.cs]: Change `string[]`  for `params string[]`

File: Flepper.QueryBuilder/Commands/SelectCommand.cs [hunk 6]
```diff
@@ -24,8 +24,13 @@ public ISelectCommand SelectCommand(params SqlColumn[] columns)
...
+	    {
+			var sqlColumns = columns.Select(c => new SqlColumn(c)).ToArray();
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/SelectCommand.cs]: Inline variable

File: Flepper.QueryBuilder/FlepperQueryBuilder.cs [hunk 7]
```diff
@@ -23,12 +23,20 @@ public static ISelectCommand Select()
...
-        /// </summary>
-        /// <typeparam name="T">Object</typeparam>
-        /// <returns></returns>
-        public static ISelectCommand Select<T>() where T : class
+	    /// <summary>
+	    /// Create Select Command
+	    /// </summary>
+	    /// <param name="columns">Columns name</param>
+	    /// <returns></returns>
+	    public static ISelectCommand Select(string[] columns)
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/FlepperQueryBuilder.cs]: Change `string[]`  for `params string[]`

File: Flepper.Tests.Unit/QueryBuilder/Commands/InsertCommandTests.cs [hunk 8]
```diff
@@ -29,7 +29,19 @@ public void ShouldCreateInsertStatementWithColumns()
                 .Be("INSERT INTO [Test] ([column1],[column2] )");
         }
 
-        [Fact]
+	    [Fact]
+	    public void ShouldCreateInsertStatementWithColumnsArray()
+	    {
+		    FlepperQueryBuilder
+			    .Insert().Into("Test")
+			    .Columns(new[] {"column1", "column2"})
```
Comment (AlbertoMonteiro) [inline Flepper.Tests.Unit/QueryBuilder/Commands/InsertCommandTests.cs]: Change `new [] {"column1", "column2"}` for `"column1", "column2"`

File: Flepper.Tests.Unit/QueryBuilder/Commands/SelectCommandTests.cs [hunk 9]
```diff
@@ -35,7 +35,18 @@ public void ShouldCreateSelectStatementWithSpecificColumns()
                 .Be("SELECT [Id],[Name],[Birthday] FROM [user]");
         }
 
-        [Fact]
+	    [Fact]
+	    public void ShouldCreateSelectStatementWithSpecificColumnsArray()
+	    {
+		    FlepperQueryBuilder.Select(new[] {"Id", "Name", "Birthday"})
```
Comment (AlbertoMonteiro) [inline Flepper.Tests.Unit/QueryBuilder/Commands/SelectCommandTests.cs]: Change `new[] {"Id", "Name", "Birthday"}` for `"Id", "Name", "Birthday"`

File: Flepper.QueryBuilder/FlepperQueryBuilder.cs (hunk 7 above)
Comment (ctyar) [reply Flepper.QueryBuilder/FlepperQueryBuilder.cs]: This change makes some of the tests fail to build:
`The call is ambiguous between the following methods or properties: 'FlepperQueryBuilder.Select(params SqlColumn[])' and 'FlepperQueryBuilder.Select(params string[])'
`

**Comment by ctyar** [comment]
The change you requested makes some of the tests fail to build:
`The call is ambiguous between the following methods or properties: 'FlepperQueryBuilder.Select(params SqlColumn[])' and 'FlepperQueryBuilder.Select(params string[])'`

Right now we are able to call these APIs with `param` like syntax:
`FlepperQueryBuilder.Select("Id", "Name", "Birthday")`
The reason for this PR was to be able to send an array directly to these methods.

**Comment by AlbertoMonteiro** [comment]
Fine, then keep with it :D
________________________________
De: Shahriar Gholami <notifications@github.com>
Enviado: sábado, 3 de março de 2018 15:12
Para: Flepper/flepper
Cc: Alberto Monteiro; Comment
Assunto: Re: [Flepper/flepper] Add array overloads to IInsertCommand and ISelectCommand (#86)


The change you requested makes some of the tests fail to build:
The call is ambiguous between the following methods or properties: 'FlepperQueryBuilder.Select(params SqlColumn[])' and 'FlepperQueryBuilder.Select(params string[])'

Right now we are able to call these APIs with param like syntax:
FlepperQueryBuilder.Select("Id", "Name", "Birthday")
The reason for this PR was to be able to send an array directly to these methods.

**Pull request merged**